### PR TITLE
 Improve cache and local bundles

### PR DESCRIPTION
**Pull request by Xiaochao8**

### CODE REVIEW

File: consts.go [hunk 1]
```diff
@@ -7,22 +7,29 @@ package sgtn
 
 // api url
 const (
-	i18nApiroot          = "/i18n/api/v2"
```
Comment (tigershi) [inline consts.go]: const,  why not use upper case

File: bundle_dao.go [hunk 2]
```diff
@@ -66,8 +84,8 @@ func (d *bundleDAO) getLocales() ([]string, error) {
 	return lSlice, nil
 }
 
-func (d *bundleDAO) getComponentMessages(locale, component string) (ComponentMsgs, error) {
-	compDirPath := filepath.Join(d.cfg.LocalBundles, component)
+func (d *bundleDAO) GetComponentMessages(name, version, locale, component string) (ComponentMsgs, error) {
```
Comment (tigershi) [inline bundle_dao.go:87]: why method return struct，not return the struct object point

Comment (Xiaochao8) [reply bundle_dao.go:87]: ComponentMsgs is an interface storing a pointer. So it returns a pointer in fact.

File: translation_interface.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package sgtn
+
+// Translation interface of translation
+type Translation interface {
```
Comment (tigershi) [inline translation_interface.go:9]: How about change the name Translation to TranslationReader

File: bundle_dao.go (hunk 2 above)
Comment (tigershi) [reply bundle_dao.go:87]: Ok, I have found it

**Pull request merged**